src/modules/users/user.repository.ts

```diff
@@ -1,12 +1,26 @@
-// В разработке :)
-// export const UserRepository = {
-//   postRegistration(dto: LoginPasswordDto, user: LoginPasswordDto) {
-//     const hash = hashSync(dto.login, appConfig.passwordRounds);
-//     if (compareSync(dto.password, user.password)) {
-//       console.log('Успех');
-//     }
-//   },
-//
-//   getLogin(id: string) {},
-//   getProfile(id: string) {},
-// };
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+import { User } from './user.types';
+
+let storage: User[] = [];
+const filename = 'users.json';
+
+const saveStorageToFile = () => writeFileSync(filename, JSON.stringify(storage, null, 2));
+
+if (existsSync(filename)) {
+  storage = JSON.parse(readFileSync(filename, 'utf-8'));
+} else {
+  saveStorageToFile();
+}
+
+export const UserRepository = {
+  registration(dto: Omit<User, 'id'>): User {
```

Comment:
```suggestion
  save(dto: Omit<User, 'id'>): User {
```
`registration` - это описание бизнес-процесса, "регистрация пользователя", прочитай ещё раз слои. База данных ничего не знает о наших бизнес-процессах. База данных знает об операциях записи, чтения, удаления и обновления. Поэтому лучшее имя - save / create / insert.